src/build/__main__.py

```diff
@@ -125,65 +130,88 @@ def main_parser():  # type: () -> argparse.ArgumentParser
         help='does not check for the dependencies',
     )
     parser.add_argument(
+        '--config-setting',
+        '-C',
+        action='append',
+        metavar='k=v',
+        type=config_setting,
+        help='pass option to the backend',
+        default=[],
+    )
+
+    group = parser.add_argument_group('isolation options')
+    default_isolation = Isolation()
+    group = group.add_mutually_exclusive_group()
+    group.add_argument(
         '--no-isolation',
         '-n',
         action='store_true',
         help='do not isolate the build in a virtual environment',
+        default=not default_isolation.enabled,
     )
-    parser.add_argument(
-        '--config-setting',
-        '-C',
-        action='append',
-        help='pass option to the backend',
+    group.add_argument(
+        '--ensurepip',
+        dest='ensure_pip',
+        action='store_true',
+        help='isolate in a virtual environment and call a fresh ensurepip every time',
+        default=default_isolation.ensure_pip,
+    )
+    group.add_argument(
+        '--cache',
+        dest='cache',
+        help='cache isolation environment(s) in between runs',
+        default=default_isolation.cache,
+    )
+    group.add_argument(
+        '--reset-cache',
+        dest='reset_cache',
+        help='clear out the tools cache',
+        action='store_true',
+        default=default_isolation.reset_cache,
```

Comment:
I don't like exposing this. This leaks implementation details all over.

Reply:
I think we should only have `--reset-tool-cache` to remove up the pip environment. We can also have a `PYTHON_BUILD_NO_CACHE` environment variable if that's necessary, but I doubt it. The argument to use an env var instead of a CLI option would be that if you are in one of those special cases where you don't want pip to be cached, you probably want to do it for all calls, for eg. in a CI, you would only set the env var one time.